src/analysis/fourier/fourier_transform.lean

```diff
@@ -0,0 +1,226 @@
+Authors: David Loeffler
+-/
+
+import analysis.special_functions.complex.circle
+import measure_theory.group.integration
+
+/-!
+# The Fourier transform
+
+We set up the Fourier transform for complex-valued functions on finite-dimensional spaces.
+
+## Design choices
+
+We define Fourier transforms in the following context:
+* `𝕜` is a commutative ring.
+* `V` and `W` are `𝕜`-modules.
+* `e` is a unitary additive character of `𝕜`, i.e. a homomorphism `(multiplicative 𝕜) →* circle`.
+* `μ` is a measure on `V`.
+* `L` is a `𝕜`-bilinear form `V × W → 𝕜`.
+* `E` is a complete normed `ℂ`-vector space.
+
+With these definitions, the Fourier transform is a map from functions `V → E` to
+functions `W → E`, defined by sending `f` to
+
+`λ w, ∫ v in V, e [-L v w] • f v ∂μ`,
+
+where `e [x]` is notational sugar for `(e (multiplicative.of_add x) : ℂ)` (available in locale
+`fourier_transform`).
+
+The most familiar case, of course, is
+* `𝕜 = V = W = ℝ`
+* `e = fourier_char` (defined below), i.e. the character `λ x, exp (2 * π * I * x)`
+* `μ = volume`
+* `L = linear_map.mul ℝ ℝ`
+
+The notation `𝓕` is available in the locale `fourier_transform` as a shortcut for this case.
+However, we set things up much more generally (number theorists care about the case when `𝕜` is an
+adele ring, for instance).
+
+## Main results
+
+At present the only nontrivial lemma we prove is `continuous_fourier_transform`, stating that the
+Fourier transform of an integrable function is continuous (under mild assumptions).
+-/
+
+noncomputable theory
+
+local notation `𝕊` := circle
+
+open measure_theory filter
+
+open_locale topology
+
+-- To avoid messing around with multiplicative vs. additive characters, we make a notation.
+localized "notation e `[` x `]` := (e (multiplicative.of_add x) : ℂ)" in fourier_transform
+
+section defs
+
+variables
+  {𝕜 : Type*} [comm_ring 𝕜]
+  {V : Type*} [add_comm_group V] [module 𝕜 V] [measurable_space V]
+  {W : Type*} [add_comm_group W] [module 𝕜 W]
+  {E : Type*} [normed_add_comm_group E] [complete_space E] [normed_space ℂ E]
+
+/-- The Fourier transform of `f : V → E`, with respect to a bilinear form `L : V × W → 𝕜` and an
+additive character `e`. -/
+def fourier_transform
+  (e : (multiplicative 𝕜) →* 𝕊) (μ : measure V) (L : V →ₗ[𝕜] W →ₗ[𝕜] 𝕜)
+  (f : V → E) (w : W) : E :=
+∫ v, e [-L v w] • f v ∂μ
+
+lemma fourier_transform.smul_const
+  (e : (multiplicative 𝕜) →* 𝕊) (μ : measure V) (L : V →ₗ[𝕜] W →ₗ[𝕜] 𝕜)
+  (f : V → E) (r : ℂ) :
+  fourier_transform e μ L (r • f) = r • (fourier_transform e μ L f) :=
+begin
+  ext1 w,
+  simp only [pi.smul_apply, fourier_transform, smul_comm _ r, integral_smul],
+end
+
+/-- The uniform norm of the Fourier transform of `f` is bounded by the `L¹` norm of `f`. -/
+lemma fourier_transform_norm_le {e : (multiplicative 𝕜) →* 𝕊} {μ : measure V}
+  (L : V →ₗ[𝕜] W →ₗ[𝕜] 𝕜) {f : V → E} (hf : integrable f μ) (w : W) :
+  ‖fourier_transform e μ L f w‖ ≤ ‖mem_ℒp.to_Lp f (mem_ℒp_one_iff_integrable.mpr hf)‖ :=
```

Comment:
This doesn't actually work because `integrable.to_L1` wants `f` as an explicit argument, despite the fact that it is trivially inferrable from hf. I'll make a separate PR to fix this.

Reply:
This doesn't work: one needs to enter it as `‖hf.to_L1 f‖`, because `integrable.to_L1` has the function as an explicit argument. But that's stupid, so I made #18435 to fix it.